roles/wait_for_active_he_vm/files/wait_for_active_he_vm

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env python
+
+import time
+from ovirtsdk.api import API
+
+
+def main():
+    args = parse_args()
+    engine_api = get_api(
+        url=args.url,
+        user=args.username,
+        password=args.password,
+        insecure=True
+    )
+    wait_for_he_vm_up_state(engine_api)
+
+
+def parse_args():
+    import argparse
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--url', dest='url', required=True,
+                        help='the url for the Engine api')
+    parser.add_argument('--username', dest='username', required=True,
+                        help='the username for the Engine api')
+    parser.add_argument('--password', dest='password', required=True,
+                        help='the password for the Engine api')
+    args = parser.parse_args()
+    return args
+
+
+def get_api(url=None, user=None, password=None, insecure=False, timeout=30):
+    start = time.time()
+    while time.time() - start < timeout:
+        try:
+            return API(
+                url=url,
+                username=user,
+                password=password,
+                insecure=insecure
+            )
+        except Exception:
+            time.sleep(10)
+
+
+def wait_for_he_vm_up_state(engine_api):
+    timeout = 600
+    vm_name = 'HostedEngine'
+    start = time.time()
+    while time.time() - start < timeout:
+        try:
+            hosted_vm = engine_api.vms.get(name=vm_name)
+            if hosted_vm.status.state == 'up':
```

Comment:
Is it enough, or do you wish to wait for the service as well (you can query the health service). Here's my code to do it:
```
HOSTEDENGINE="lago-he-basic-suite-4-1-engine"
DOMAIN=`dnsdomainname`

counter=100
while [ $counter -gt 0 ]
do
    code=`curl --insecure --silent --output /dev/null --write-out '%{http_code}' https://${HOSTEDENGINE}.${DOMAIN}/ovirt-engine/services/health`
    if [ "${code}" == 200 ]; then
        break
    fi
    counter=$(( $counter - 1 ))
    sleep 3
done
if [ ${counter} -eq 0 ]; then
    echo "Could not verify HE health"
    exit 1
fi
```